### PR TITLE
refactor: rewrite YearOverYearSelect component (TECH-321)

### PR DESCRIPTION
**Pull request by edoardo**
Fixes: https://jira.dhis2.org/browse/TECH-321

Part of the effort for reducing the technical debt in DV.

Rewrite the `YearOverYearSelect` component (under `components/Layout/YearOverYearLayout`) to use `ui-core` in place of material UI.

For the series axis the `MultiSelect` is used, while for the category axis `SingleSelect`.

The look is different from before.

Before:
Screenshot 2020-04-02 at 16 11 39

After:
Screenshot 2020-04-15 at 15 06 48

@cooper-joe ~is working~ worked on a `dense` version of the `Chip` to save some space.

Things left to do:

1. review the styling and do necessary adjustments to accommodate better the new components
    - Spoke with @cooper-joe about this and we agreed it's ok like it is now, with no additional style tweaking, otherwise we should use a custom component
2. test with lots of item selected, currently this is what it looks like:
Screenshot 2020-04-15 at 15 08 53

### CODE REVIEW

**Comment by edoardo** [comment]
Waiting for this to be merged: https://github.com/dhis2/ui-core/pull/875.

**Comment by martinkrulltott** [comment]
Discussion on Slack: https://dhis2.slack.com/archives/G0108CD1421/p1586956390007200

**Comment by martinkrulltott** [comment]
@cooper-joe asked for the width of the select fields to be 100%. This is blocked by https://github.com/dhis2/ui/issues/39

**Comment by janhenrikoverland** [comment]
@martinkrulltott Regarding layout rules for YOY, we are doing this: https://github.com/dhis2/data-visualizer-app/blob/master/packages/app/src/modules/layoutValidation.js#L76-L82 which _knows_ that `yearlySeries` is series in the ui. We should ideally refer to the rules instead, but for now it works without any changes to the rules.

File: yarn.lock [hunk 1]
```diff
@@ -1748,19 +1748,26 @@
...
   resolved "https://registry.yarnpkg.com/@dhis2/prop-types/-/prop-types-1.5.0.tgz#7e69919f66698be373dd21940a8a770234ded6a1"
   integrity sha512-dueFkkAMOIMbXiU7Mhr3Y+DBRyOd/rHA+5/IDiYWN1xttlUTSuGZLQ5AnJ7osBicEhx+qElaGbTdRYQj3SMBtA==
   dependencies:
     prop-types "^15"
 
-"@dhis2/ui-core@^4.16.0", "@dhis2/ui-core@^4.6.1", "@dhis2/ui-core@^4.9.1":
-  version "4.16.0"
-  resolved "https://registry.yarnpkg.com/@dhis2/ui-core/-/ui-core-4.16.0.tgz#868555c2d8758af3f26e168d2a469bde39c91b1d"
-  integrity sha512-pa47/ZZrQa9SUXlOW4r8HKyuzfO/i4vENR1u2+2zrPfBtbFK4FPwXANmdVAw+V6epbWJU8k2Fi31j1ZMwauq/w==
+"@dhis2/prop-types@^1.6.4":
```
Comment (martinkrulltott) [inline yarn.lock:1758]: Looks like we're getting duplicate versions of `@dhis2/prop-types`?

Comment (amcgee) [reply yarn.lock:1758]: Duplicate @dhis2/prop-types should be VERY small and not a singleton, so I don't think it's a huge deal here but good to keep an eye on `yarn.lock` changes!

**Pull request merged**